aloha-core/src/main/scala/com/eharmony/aloha/dataset/RowCreatorProducer.scala

```diff
@@ -5,15 +5,19 @@ import spray.json.JsValue
 import com.eharmony.aloha.semantics.compiled.CompiledSemantics
 
 /**
- * SpecProducer is used to create different kinds of [[RowCreator]] instances.  '''Classes that extend SpecProducer should
- * only have zero-arg constructors.'''  This is because Spec instances should only be parametrized by the JSON
- * specification.  Otherwise, one JSON specification could produce non-equivalent Spec instances in different
- * environments.  It is a design goal for this not to happen.
- *
- * @tparam A type of input passed to the spec.
- * @tparam B implementation of the Spec[A] that is returned by the getSpec function.
- */
-trait RowCreatorProducer[A, +B <: RowCreator[A]] {
+  * [[RowCreatorProducer]] is used to create different kinds of [[RowCreator]] instances.
+  * '''Classes that extend [[RowCreatorProducer]] should only have zero-arg constructors.'''
+  * This is because Spec instances should only be parametrized by the JSON
+  * specification.  Otherwise, one JSON specification could produce non-equivalent
+  * Spec instances in different environments.
+  *
+  * '''It is a design goal for this not to happen.'''
```

Comment:
Update this comment for the multi-label.  Talk about the unit test and what exceptions to the rule might exist.